src/dmd/root/longdouble.d

```diff
@@ -6,29 +6,30 @@
  * https://github.com/dlang/dmd/blob/master/src/root/longdouble.d
  */
 
-// 80 bit floating point value implementation for LDC compiler targetting MSVC
+// 80-bit floating point value implementation if the C/D compiler do not support them natively
 
 module dmd.root.longdouble;
 
-enum hasNativeFloat80 = real.sizeof > 8;
-
-static if (hasNativeFloat80)
-{
+static if (real.sizeof > 8)
     alias longdouble = real;
-}
+else
+    alias longdouble = longdouble_soft;
 
-static if (!hasNativeFloat80):
+// longdouble_soft needed when building the backend with
+// Visual C or the frontend with LDC on Windows
+version(CRuntime_Microsoft):
 extern(C++):
 nothrow:
+@nogc:
 
-align(2) struct longdouble
+align(2) struct longdouble_soft
 {
-nothrow:
+nothrow @nogc:
     ulong mantissa = 0xC000000000000001UL; // default to snan
```

Comment:
I gave up on snan a while ago. Nothing supports it, and it just causes problems when people compare nan bits. Use qnan instead.

Reply:
Ok, updated.